bubuku/features/restart_if_dead.py

```diff
@@ -8,48 +8,47 @@
 
 _LOG = logging.getLogger('bubuku.features.restart_if_dead')
 
-
 class CheckBrokerStopped(Check):
     def __init__(self, broker: BrokerManager, zk: BukuExhibitor):
         super().__init__()
         self.broker = broker
         self.zk = zk
         self.need_check = True
-        self.last_zk_session_check = datetime.now()
+        self.last_zk_session_failed_check = None
 
     def check(self) -> Change:
         if not self.need_check:
             return None
-        if self.is_running_and_registered(should_retry=True):
+        if not self.should_restart():
             return None
 
         _LOG.warning('Oops! Broker is dead, triggering restart')
         self.need_check = False
 
         # Do not start if broker is running and registered
         def _cancel_if():
-            return self.is_running_and_registered()
+            return not self.broker.is_running() or self.broker.is_registered_in_zookeeper()
 
         return RestartBrokerChange(self.zk, self.broker, _cancel_if, self.on_check_removed)
 
     # Attempt to verify that broker is not registered in zookeeper for twice as long as the zookeeper session timeout.
     # Allow zookeeper client to try to restore the session before killing tha kafka process as soon as zookeeper session is dead.
-    def is_running_and_registered(self, should_retry=False):
+    def should_restart(self):
         current_time = datetime.now()
         if not self.broker.is_running():
-            return False
+            return True
         if not self.broker.is_registered_in_zookeeper():
             _LOG.warning('Broker is not regiestered in Zookeeper')
-            time_to_return = self.last_zk_session_check + timedelta(milliseconds=self.broker.get_zookeeper_session_timeout() * 2)
-            if current_time > time_to_return or not should_retry:
-                self.last_zk_session_check = current_time
-                return False
-        else:
-            self.last_zk_session_check = current_time
-        return True
+            if not self.last_zk_session_failed_check:
+                self.last_zk_session_failed_check = current_time
+            time_to_restart = self.last_zk_session_failed_check + timedelta(milliseconds=self.broker.get_zookeeper_session_timeout() * 2)
```

Comment:
Maybe it is only a matter of taste, but I would prefer to name `time_to_restart` as `time_to_restart_at`